workspace/lib/lita/handlers/teamcity.rb

```diff
@@ -337,6 +350,27 @@ def artifacts(response, have_build_number)
         end
       end
 
+      def artifactory(response)
+        pr_id = response.match_data['pr_id']
+        result = ""
+        begin
+          full_url, specific_url = fetch_download_url_by_pr_id(pr_id)
```

Comment:
I think we should handle exception on inside this method.

Reply:
In case handle exception on inside this method, I couldn't find the way to test unittest for fetch_download_url_by_pr_id with `response`. Please let me know if there is any way to do it.
Besides,
- if handling exception on inside `fetch_download_url_by_pr_id` method, some other actions/methods are still running after the exception. And it's not good.
- if handling exception on inside `artifactory` method, the artifactory will be stopped immediately.

Reply:
got it